src/app/home/_component/mission_badge.tsx

```diff
@@ -0,0 +1,38 @@
+import { MoonIcon } from '@heroicons/react/20/solid';
+import { TicketIcon } from '@heroicons/react/20/solid';
+import { RocketLaunchIcon } from '@heroicons/react/20/solid';
+
+import Badge from '@/components/ui/badge';
+
+type Mission_type = 'DAILY' | 'WEEKLY' | 'SPECIAL';
+type Props = {
+  missionType: Mission_type;
+};
+
+export default function MissionBadge({ missionType }: Props) {
+  const missions = {
```

Comment:
이 타입 자체에 대해 조금 더 엄격하게 검증을 하는게 좋아보입니다.


```suggestion
type Mission_type = 'DAILY' | 'WEEKLY' | 'SPECIAL';

type MissionInfo = {
  icon: ComponentType<SVGProps<SVGSVGElement>>;
  text: string;
  color: 'primary' | 'secondary' | 'success' | 'warning' | 'error' | 'white';
};

const missions: Record<MissionType, MissionInfo> = {
  DAILY: {
    icon: MoonIcon,
    text: '일간 미션',
    color: 'white',
  },
  WEEKLY: {
    icon: TicketIcon,
    text: '주간 미션',
    color: 'white',
  },
  SPECIAL: {
    icon: RocketLaunchIcon,
    text: '특별 미션',
    color: 'success',
  },
};
```

Reply:
`<MoonIcon className="w-3 h-3" />` 대신 `MoonIcon`과 같이 컴포넌트 타입만 정의한 이유는 크기를 전체적으로 조정하고 싶을 때, MissionType에 대해 3개의 경우를 바꾸는 것 보다, 렌더링 당시에 한 줄의 코드만 수정해도 되기 때문입니다.

Reply:
추가로 저 color들이 계속 쓰일 것 같아서, `app/types/colors.ts` 와 같은 곳에 별도로 정의하는게 더 좋아보입니다.